fetch-in-react-practice/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+/* eslint-disable @typescript-eslint/no-empty-function -- Remove me */
+import { useEffect, useState } from 'react';
+import { PageTitle } from './PageTitle';
+import { TodoList } from './TodoList';
+import { TodoForm } from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function getTodos() {
+      try {
+        const response = await fetch('/api/todos');
+        const data = await response.json();
+        if (!response.ok) {
+          throw new Error(`Error: ${response.status}`);
+        }
+        setTodos(data);
+        setIsLoading(false);
+      } catch (error) {
+        setError(error);
```

Comment:
setIsLoading(false) on his branch also, or you won't ever display the error message.